guacamole-common/src/main/java/org/apache/guacamole/protocol/ConfiguredGuacamoleSocket.java

```diff
@@ -142,6 +150,13 @@ public ConfiguredGuacamoleSocket(GuacamoleSocket socket,
 
             // Retrieve argument name
             String arg_name = arg_names.get(i);
+            
+            // Check for protocol version as first argument
+            if (i == 0 && arg_name.startsWith("VERSION_")) {
+                protocol = GuacamoleProtocolVersion.valueOf(arg_name);
```

Comment:
As `valueOf()` for an enum is specifically defined based on the names of the enum values, I'm uncomfortable directly relying on it for parsing of the Guacamole protocol (something which technically is independent of the Java objects we may use to implement it).

I suggest providing a function within `GuacamoleProtocolVersion` which is specifically defined as parsing the Guacamole protocol constant for that protocol version and returning the corresponding enum value. Even though that function may internally only wrap `valueOf()`, it will decouple the internals of the API from the internals of the protocol.

Reply:
Okay, I implemented a `getVersion()` method that checks for null/empty and then wraps `valueOf()` for the time being.  I'm happy to do something different in `getVersion()` if that's desirable, but at least it decouples from reliance upon `valueOf()` at this point.